src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/FlowPathSwapHubBolt.java

```diff
@@ -0,0 +1,151 @@
+import org.openkilda.messaging.command.flow.FlowPathSwapRequest;
+import org.openkilda.messaging.command.flow.PeriodicPingCommand;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.wfm.error.PipelineException;
+import org.openkilda.wfm.share.flow.resources.FlowResourcesConfig;
+import org.openkilda.wfm.share.flow.resources.FlowResourcesManager;
+import org.openkilda.wfm.share.history.model.FlowHistoryHolder;
+import org.openkilda.wfm.share.hubandspoke.HubBolt;
+import org.openkilda.wfm.share.utils.KeyProvider;
+import org.openkilda.wfm.topology.flowhs.FlowHsTopology.Stream;
+import org.openkilda.wfm.topology.flowhs.service.FlowPathSwapHubCarrier;
+import org.openkilda.wfm.topology.flowhs.service.FlowPathSwapService;
+import org.openkilda.wfm.topology.utils.MessageKafkaTranslator;
+
+import lombok.Builder;
+import lombok.Getter;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+
+public class FlowPathSwapHubBolt extends HubBolt implements FlowPathSwapHubCarrier {
+
+    private final FlowPathSwapConfig config;
+    private final PersistenceManager persistenceManager;
+    private final FlowResourcesConfig flowResourcesConfig;
+
+    private transient FlowPathSwapService service;
+    private String currentKey;
+
+    public FlowPathSwapHubBolt(FlowPathSwapConfig config, PersistenceManager persistenceManager,
+                               FlowResourcesConfig flowResourcesConfig) {
+        super(config);
+
+        this.config = config;
+        this.persistenceManager = persistenceManager;
+        this.flowResourcesConfig = flowResourcesConfig;
+
+
+    }
+
+    @Override
+    protected void init() {
+
+        FlowResourcesManager resourcesManager = new FlowResourcesManager(persistenceManager, flowResourcesConfig);
+        service = new FlowPathSwapService(this, persistenceManager, config.getTransactionRetriesLimit(),
+                config.getSpeakerCommandRetriesLimit(), resourcesManager);
+    }
+
+    @Override
+    protected void onRequest(Tuple input) throws PipelineException {
+        currentKey = input.getStringByField(MessageKafkaTranslator.FIELD_ID_KEY);
+        FlowPathSwapRequest payload = (FlowPathSwapRequest) input.getValueByField(FIELD_ID_PAYLOAD);
+        service.handleRequest(currentKey, pullContext(input), payload);
+    }
+
+    @Override
+    protected void onWorkerResponse(Tuple input) {
+        String operationKey = input.getStringByField(MessageKafkaTranslator.FIELD_ID_KEY);
+        currentKey = KeyProvider.getParentKey(operationKey);
+        SpeakerFlowSegmentResponse flowResponse = (SpeakerFlowSegmentResponse) input.getValueByField(FIELD_ID_PAYLOAD);
+        service.handleAsyncResponse(currentKey, flowResponse);
+    }
+
+    @Override
+    public void onTimeout(String key, Tuple tuple) {
+        currentKey = key;
+        // service.handleTimeout(key);
```

Comment:
commented code

Reply:
+1 
Is there any reason why this code is commented out?

Reply:
Fixed, thanks